packages/core/src/browser/shell/tab-bar-toolbar/tab-bar-toolbar.tsx

```diff
@@ -162,10 +171,14 @@ export class TabBarToolbar extends ReactWidget {
         </div>;
     }
 
+    protected isEnabled(item: AnyToolbarItem): boolean {
+        return (!!item.command && this.commandIsEnabled(item.command) && this.evaluateWhenClause(item.when)) || (!!item.menuPath && !item.command);
```

Comment:
nitpick: maybe it helps readability if we extract two variables with helpful names for each of the two parts of the conditions.